File: app/src/main/java/com/example/networkaplication/persistance/model/MovieQuery.java
package com.example.networkaplication.persistance.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import java.util.Objects;

@Entity(tableName = "movie_query")
public class MovieQuery {

    @PrimaryKey(autoGenerate = true)
    private long id;
    private String name;
    private long date;

    public MovieQuery () {}

    @Ignore
    public MovieQuery (String name, long date) {
        this.name = name;
        this.date = date;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MovieQuery that = (MovieQuery) o;
        return id == that.id &&
                date == that.date &&
                Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, date);
    }

}

File: app/src/main/java/com/example/networkaplication/persistance/model/MovieQuery.kt
package com.example.networkaplication.persistance.model

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

import java.util.Objects

@Entity(tableName = "movie_query")
class MovieQuery {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    var name: String? = null
    var date: Long = 0

    constructor() {}

    @Ignore
    constructor(name: String, date: Long) {
        this.name = name
        this.date = date
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as MovieQuery?
        return id == that!!.id &&
                date == that.date &&
                name == that.name
    }

    override fun hashCode(): Int {
        return Objects.hash(id, name, date)
    }
}
